gsdk/src/signer/mod.rs

```diff
@@ -17,22 +17,55 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 //! Gear api with signer
+
 use crate::{
     config::GearConfig,
     result::{Error, Result},
     Api,
 };
+use core::ops::Deref;
 pub use pair_signer::PairSigner;
 use sp_core::{crypto::Ss58Codec, sr25519::Pair, Pair as PairT};
 use sp_runtime::AccountId32;
+use std::sync::Arc;
 
 mod calls;
 mod pair_signer;
 mod rpc;
 mod utils;
 
+/// Signer representation that provides access to gear API.
+/// Implements low-level methods such as [`run_tx`](`SignerInner::run_tx`)
+/// and [`force_batch`](`SignerInner::force_batch`).
+/// Other higher-level calls are provided by [`Signer::sudo`],
+/// [`Signer::balance`], [`Signer::calls`], [`Signer::rpc`].
 #[derive(Clone)]
 pub struct Signer {
+    signer: Arc<SignerInner>,
+    /// Calls that require sudo.
+    pub sudo: SignerSudo,
+    /// Calls to interact with account balance.
+    pub balance: SignerBalance,
+    /// Calls for interaction with on-chain programs.
+    pub calls: SignerCalls,
```

Comment:
sudo and balances could be merged in to calls

Reply:
Even if I move `transfer` to the calls then there still remains `get()` and `log_spent()` fns. And why `sudo` should be merged into calls? These calls are belong to different use-cases and the border between them is obvious

Reply:
how do you feel about
* ...pallets
    * calls
    * storage
* rpc

Reply:
> how do you feel about

hmmm, feeling this discussion is endless, but sure we can write some macro to implement it, I'm more closed to 

storage.pallet_name.query because it is synced with the polkadot.js API

Reply:
> currently, imo just introducing calls/storage/rpc is enough ))

where do I put `log_balance_spent()` then?

Reply:
> storage.pallet_name.query

But why do we need to let user know about pallet existence?

Reply:
> where do I put log_balance_spent() then?

I got you now, mb we can move this method to `Singer` or just give it to `gclient`

```rust
struct Signer {
  inner
}
```

because Signer has the access to the pallet balances and all of other calls